lib/codecs/src/decoding/format/syslog.rs

```diff
@@ -101,14 +110,88 @@ impl SyslogDeserializerConfig {
+                        &owned_value_path!("timestamp"),
+                        Kind::timestamp(),
+                        Some("timestamp"),
+                    )
+                    .with_source_metadata(
+                        source,
+                        None,
+                        &owned_value_path!("hostname"),
+                        Kind::bytes().or_undefined(),
+                        None,
+                    )
+                    .with_source_metadata(
+                        source,
+                        None,
+                        &owned_value_path!("severity"),
+                        Kind::bytes().or_undefined(),
+                        Some("severity"),
+                    )
+                    .with_source_metadata(
+                        source,
+                        None,
+                        &owned_value_path!("facility"),
+                        Kind::bytes().or_undefined(),
+                        None,
+                    )
+                    .with_source_metadata(
+                        source,
+                        None,
+                        &owned_value_path!("version"),
+                        Kind::integer().or_undefined(),
+                        None,
+                    )
+                    .with_source_metadata(
+                        source,
+                        None,
+                        &owned_value_path!("appname"),
+                        Kind::bytes().or_undefined(),
+                        None,
+                    )
+                    .with_source_metadata(
+                        source,
+                        None,
+                        &owned_value_path!("msgid"),
+                        Kind::bytes().or_undefined(),
+                        None,
+                    )
+                    .with_source_metadata(
+                        source,
+                        None,
+                        &owned_value_path!("procid"),
+                        Kind::integer().or_bytes().or_undefined(),
+                        None,
+                    )
+                    .with_source_metadata(
+                        source,
+                        None,
+                        &owned_value_path!("structured_data"),
+                        Kind::object(Collection::from_unknown(Kind::object(
+                            Collection::from_unknown(Kind::bytes()),
+                        ))),
+                        None,
+                    )
+            }
         }
     }
 }
 
 /// Deserializer that builds an `Event` from a byte frame containing a syslog
 /// message.
 #[derive(Debug, Clone)]
-pub struct SyslogDeserializer;
+pub struct SyslogDeserializer {
+    /// The syslog source needs it's own syslog deserializer separate from the
+    /// syslog codec since it needs to handle the structured of the decoded data
+    /// differently when using the Vector lognamespace.
+    pub source: Option<&'static str>,
```

Comment:
Reading this code without the overall context, the naming here doesn't feel great. Any decoder configured will be associated with a source, that said I don't know if I have a better naming idea to make it more clear but I'm not sold on this currently.

Reply:
Should it just be a boolean and have the `syslog` source name hardcoded internally? That's the only time this added behavior would apply, and may be less prone to misuse?